### ISSUE TITLE
Make enums exotic types that throw on unknown property access

### ISSUE THREAD

**Issue by domenic** [rwaldron/enum-definitions#2]
As stated, there is little benefit to the proposal over simply using strings: instead of typing `Color.BLUE`, you could just type `"BLUE"`. Maybe the proposal helps in IDE scenarios where you have autocomplete, but that's not a big enough win IMO.

Instead, I think the real win would come if `Color.BLU`, or any other typo, threw an error. This would require defining an exotic [[Get]] on enum objects, I believe, instead of just using `Object.create(null)`.

**Comment by rwaldron** [comment]
Yes! I had this written in my notebook, glad you mentioned it here as well (ensure that it doesn't get forgotten in said notebook)
